src/BugsnagServiceProvider.php

```diff
@@ -328,6 +334,40 @@ protected function setupPaths(Client $client, $base, $path, $strip, $project)
         $client->setProjectRoot($path);
     }
 
+    /**
+     * Setup session tracking.
+     *
+     * @param \Bugsnag\Client $client
+     * @param string          $endpoint
+     *
+     * @return void
+     */
+    protected function setupSessionTracking(Client $client, $endpoint)
+    {
+        $client->setSessionTracking(true, $endpoint);
+        $sessionTracker = $client->getSessionTracker();
+
+        $sessionStorage = function ($session = null) {
+            if (is_null($session)) {
+                return session('bugsnag-session', []);
+            } else {
+                session(['bugsnag-session' => $session]);
+            }
+        };
+
+        $sessionTracker->setSessionFunction($sessionStorage);
+
+        $genericStorage = function ($key, $value = null) {
+            if (is_null($value)) {
+                return Cache::get($key, null);
```

Comment:
You'll want `$this->app->cache->get(...)`

Reply:
The [laravel docs](https://laravel.com/docs/5.4/cache) suggest accessing the Cache through static methods

Reply:
Yeh, though those docs are not for package developers.

Reply:
If you go into the `config/app.php` file in the example app and delete the facade alias for `Cache`, you will be able to observe that the service provider code here doesn't work.